### PR TITLE
Lightning Transformers Auto-regressive LM Pre-training

### PR DESCRIPTION
**Pull request by footprinthere**

### CODE REVIEW

File: examples/pth-lightning/arlm-pretraining/main.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import os
+from tqdm import tqdm
+
+
+@record
+def main(args: argparse.Namespace):
+    os.environ["TOKENIZERS_PARALLELISM"] = "false"
+    os.environ["CUDA_LAUNCH_BLOCKING"] = "1"
+
+    torch_ddp.init_process_group(backend='nccl')
```
Comment (ktaebum) [inline examples/pth-lightning/arlm-pretraining/main.py:23]: Do we need explicit call of `torch_ddp`? From the official [document](https://pytorch-lightning.readthedocs.io/en/latest/accelerators/gpu_intermediate.html), we could enable torch ddp by
```
trainer = Trainer(accelerator="gpu", devices=8, strategy="ddp", num_nodes=4)
```

**Comment by footprinthere** [comment]
Edited to reflect your comment.

**Comment by ktaebum** [comment]
Thx :) Could you also apply the diff to https://github.com/friendliai/periflow-python-sdk/pull/44?

File: examples/pth-lightning/arlm-pretraining/main.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import periflow_sdk as pf
+import argparse
+import os
+from tqdm import tqdm
+
+
+@record
+def main(args: argparse.Namespace):
+    os.environ["TOKENIZERS_PARALLELISM"] = "false"
+    os.environ["CUDA_LAUNCH_BLOCKING"] = "1"
```
Comment (ktaebum) [inline examples/pth-lightning/arlm-pretraining/main.py:20]: Should we set this environment variable? It is for debugging purpose

**Comment by footprinthere** [comment]
Reflected, also in the other PR.
Next time I will double check if there are remaining unnecessary lines!

**Pull request merged**